source/upgrade.txt

```diff
@@ -0,0 +1,156 @@
+.. _java-upgrade-driver:
+
+=======================
+Upgrade Driver Versions
+=======================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+
+Overview
+--------
+
+In this section, you can identify the changes you need to make to your
+application to upgrade your driver to a new version or to transition from
+the legacy API to the current API.
+
+When you upgrade your driver version or use multiple versions of the
+driver, if there are any breaking changes between them that you have not
+addressed in the :ref:`Breaking Changes <java-breaking-changes>` section,
+you may encounter one or more of the following issues:
+
+- Your application may not compile.
+- Your application may not be compatible with data that originated from an
+  application that uses another version of the driver.
+
+.. note::
+
+   Before upgrading driver versions, check the :ref:`Java Compatibility <java-compatibility-tables>`
+   page to ensure the new version is compatible with the version of MongoDB as
+   well as the version of the Java Runtime Environment (JRE) your application
+   runs on.
+
+When you transition from the legacy API to the current API, if you use any
+of the methods or classes listed in the :ref:`Upgrade from the Legacy API <java-upgrade-from-legacy>`
+section, you need to update them to use the current methods or classes before
+removing the legacy API from your dependencies.
+
+In addition to updating your application to handle any necessary changes,
+always check for any other differences in options and return values before
+moving it to production.
+
+.. tip::
+
+   To minimize the amount of changes your application may need when
+   upgrading driver versions in the future, use the
+   :ref:`{+stable-api+} <stable-api-java>`.
+
+.. _java-breaking-changes:
+
+Breaking Changes
+----------------
+
+The breaking changes in this section are categorized by the driver version that
+introduced them.
+
+If you are either upgrading the driver version in your application or plan
+to run multiple applications that use different driver versions, address
+any of the breaking changes between the two versions. For example, if you
+are upgrading the driver from v4.0 to v4.5, address all breaking changes from
+the version after v4.0 including any listed under v4.5.
+
+.. _java-breaking-changes-v4.2:
+
+Version 4.2 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- The ``ObjectId`` class and its ``serialVersionUID`` field were updated
+  to use a new format that minimizes serialization compatibility issues
+  across different versions of the driver.
+
+  Any objects that include an ``ObjectId`` field that were serialized with an
+  older version of the driver are not deserializable by this driver version
+  and later. When the driver encounters an incompatible format, it throws an
+  exception.
+
+.. _java-upgrade-from-legacy:
+
+Upgrade from the Legacy API
```

Comment:
I don't think this belongs on an Upgrade Driver Versions page, as it's orthogonal to driver version: all driver versions support both legacy and new API.

Can we move to a new top-level page?

Reply:
Sure, I can split it into its own page.